server/storage/tuf_store_test.go

```diff
@@ -0,0 +1,106 @@
+// to the snapshot specified in the checksum, to potentially other role metadata by checksum
+func TestTUFSQLGetCurrent(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	gormDB, tufDBStore := SetupTUFSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+	defer gormDB.Close()
+
+	ConsistentEmptyGetCurrentTest(t, tufDBStore)
+
+	// put an arbitrary piece of data in the database
+	tuf := SampleTUF(1)
+	query := gormDB.Create(&tuf)
+	require.NoError(t, query.Error, "Creating a row in an empty DB failed.")
+
+	ConsistentMissingTSAndSnapGetCurrentTest(t, tufDBStore)
+
+	// Note that get by checksum succeeds, since it does not try to walk timestamp/snapshot
+	GetChecksumFoundTest(t, tufDBStore, tuf)
+
+	// Now setup a valid tuf repo and use it to ensure we walk correctly
+	validTUFRepo, _, err := testutils.EmptyRepo("testGUN")
+	require.NoError(t, err)
+
+	// Add the timestamp, snapshot, targets, and root to the database
+	tufData, err := json.Marshal(validTUFRepo.Timestamp)
+	require.NoError(t, err)
+	tsTUF := SampleCustomTUF(data.CanonicalTimestampRole, "testGUN", tufData, validTUFRepo.Timestamp.Signed.Version)
+	query = gormDB.Create(&tsTUF)
+	require.NoError(t, query.Error, "Creating a row for timestamp in DB failed.")
+
+	tufData, err = json.Marshal(validTUFRepo.Snapshot)
+	require.NoError(t, err)
+	snapTUF := SampleCustomTUF(data.CanonicalSnapshotRole, "testGUN", tufData, validTUFRepo.Snapshot.Signed.Version)
+	query = gormDB.Create(&snapTUF)
+	require.NoError(t, query.Error, "Creating a row for snapshot in DB failed.")
+
+	tufData, err = json.Marshal(validTUFRepo.Targets[data.CanonicalTargetsRole])
+	require.NoError(t, err)
+	targetsTUF := SampleCustomTUF(data.CanonicalTargetsRole, "testGUN", tufData, validTUFRepo.Targets[data.CanonicalTargetsRole].Signed.Version)
+	query = gormDB.Create(&targetsTUF)
+	require.NoError(t, query.Error, "Creating a row for targets in DB failed.")
+
+	tufData, err = json.Marshal(validTUFRepo.Root)
+	require.NoError(t, err)
+	rootTUF := SampleCustomTUF(data.CanonicalRootRole, "testGUN", tufData, validTUFRepo.Root.Signed.Version)
+	query = gormDB.Create(&rootTUF)
+	require.NoError(t, query.Error, "Creating a row for root in DB failed.")
+
+	// GetCurrent on all of these roles should succeed
+	ConsistentGetCurrentFoundTest(t, tufDBStore, tsTUF)
+	ConsistentGetCurrentFoundTest(t, tufDBStore, snapTUF)
+	ConsistentGetCurrentFoundTest(t, tufDBStore, targetsTUF)
+	ConsistentGetCurrentFoundTest(t, tufDBStore, rootTUF)
+
+	// Delete snapshot
+	query = gormDB.Delete(&snapTUF)
+	require.NoError(t, query.Error, "Deleting a row for snapshot in DB failed.")
+
+	// GetCurrent snapshot lookup should still succeed because of caching
+	ConsistentGetCurrentFoundTest(t, tufDBStore, snapTUF)
+
+	// targets and root lookup on GetCurrent should also still succeed because of caching
+	ConsistentGetCurrentFoundTest(t, tufDBStore, targetsTUF)
+	ConsistentGetCurrentFoundTest(t, tufDBStore, rootTUF)
```

Comment:
Non-blocking - would it also be useful to add another orphaned root, and show that `GetCurrent` root returns the previous root even though the orphaned root is newer?